### ISSUE TITLE
Rule identical_contents doesn't check all structure fields

### ISSUE THREAD

**Issue by pokrakam** [abaplint/abaplint#2296]
Following code generates a false positive:
```ABAP 
   DATA o TYPE hrsobject.
    IF a = b.
      o-otype = 'P'.
      o-objid = '123'.
    ELSE.
      o-otype = 'P'.
      o-objid = '456'.
    ENDIF.
```
I guess it's just looking at the first element.

**Comment by larshp** [comment]
I'd say its an expected error, the code can be refactored to

```abap
   DATA o TYPE hrsobject.
    o-otype = 'P'.
    IF a = b.
      o-objid = '123'.
    ELSE.
      o-objid = '456'.
    ENDIF.
```

**Comment by pokrakam** [comment]
Good point, but then what if there's an `ELSEIF` where one of the three is different? 
I think it may be context-driven whether it may be preferable to keep the structure together, the source where I encountered this was a range: 
```ABAP
IF ...
  r-opt = 'I'.
  r-opt = 'EQ'.
  r-low = foo.
ELSE. 
  r-opt = 'I'.
  r-opt = 'BT'.
...
ENDIF.
```
Personally I'd prefer to keep the range together, but am OK if you want to close this issue.

**Comment by larshp** [comment]
if there is an `ELSEIF`, I dont think it should report, try it out 😄 

this rule is about reducing the lines of code
